hail/src/main/scala/is/hail/annotations/StagedRegionValueBuilder.scala

```diff
@@ -65,14 +71,17 @@ class StagedRegionValueBuilder private (val mb: EmitMethodBuilder[_], val typ: P
     this(er.mb, rowType, er.region, null)
   }
 
-  private val ftype = typ.fundamentalType
+  private val stype = typ match {
+    case t: PNDArray => t.representation.fundamentalType
```

Comment:
wait, what if there's another ndarray in there? I think we need to recursively replace ndarrays with their representation, and otherwise fundamental types? Crap.

Reply:
Agh, you're right. So I could just write a function called`to_rvb_supported_type` that recursively replaces NDArray's with their structs. Alternatively, I could just make sure every function is ready to accept ndarrays instead of this current thing where I'm trying to replace all the ndarrays at the beginning.

Reply:
Decided to get rid of this `stype` business. Instead of trying to just replace the `PNDArray` with a `PStruct`, I went ahead and added a `PNDArray` case to all the match statements. No more recursion issues.